hashicorp/terraform-provider-tfe
# Guide script to use terraform-plugin-docs

## Description

A guide script for contributors to add/edit docs on the [Registry](https://registry.terraform.io/providers/hashicorp/tfe/latest) website. 

**What does it do?**
Walks users through preparing the required templates and examples before running the `tfplugindocs` command to generate markdown

**When to use?**
When users add new resources/data sources or edit pre-existing ones that require doc changes to attribute descriptions, examples, and import statements.

**How to use?**
Github PR template will include a reminder to run the script locally after making any relevant changes to the 
"tfe" directory files.

## Testing plan

1.  Go into "scripts/menu-scripts" directory 
2. Run ` ./1-main-menu.sh` to open script menu and follow steps
3. Check the "docs" directory for the markdown files corresponding to your resource
4. Paste the `.md` file in the Registry[ Docs Preview Tool](https://registry.terraform.io/tools/doc-preview) and compare how the preview looks to the existing Registry page (please comment any feedback)

In addition, if you selected the CUSTOM route, check: 
- "templates" directory for the `.tmpl` file corresponding to your resource
